File: C#/Question2_Solution2/NumberClass.cs
using System.Collections.Generic;

namespace Question2_Solution2
{
    public class NumberClass
    {
        public NumberClass(long targetNumber)
        {
            this.targetNumber = targetNumber;
            InitializeList();
        }


        private int pointerIndex;
        private readonly long targetNumber;

        private void InitializeList()
        {
            listToMultiple = new List<long> {2, 3, 5};
        }

        public long SmallestInheritor()
        {
            return targetNumber * ListToMultiple[pointerIndex];
        }

        public bool MovePointer()
        {
            pointerIndex++;
            return pointerIndex < ListToMultiple.Count;
        }

        private List<long> listToMultiple;

        public long TargetNumber => targetNumber;

        public List<long> ListToMultiple => listToMultiple;
    }
}

File: C++/Question2_Solution2/NumberClass.cpp
//
// Created by galax on 3/19/2020.
//

#include <iostream>
#include "NumberClass.h"

NumberClass::NumberClass(long targetNumber): targetNumber(targetNumber) {
    InitializeList();
}

void NumberClass::InitializeList() {
    ListToMultiple = {2,3,5};
}

long NumberClass::SmallestCombination() {
    return targetNumber * ListToMultiple[pointerIndex];
}

bool NumberClass::MovePointer() {
    pointerIndex = pointerIndex+1;
    return pointerIndex < ListToMultiple.size();
}

